vendor.conf

```diff
@@ -1,11 +1,12 @@
-github.com/agl/ed25519 d2b94fd789ea21d12fac1a4443dd3a3f79cda72c
+github.com/agl/ed25519 5312a61534124124185d41f09206b9fef1d88403
 github.com/Azure/go-ansiterm d6e3b3328b783f23731bc4d058875b0371ff8109
-github.com/beorn7/perks 4c0e84591b9aa9e6dcfdf3e020114cd81f89d5f9
+github.com/beorn7/perks 3a771d992973f24aa725d07868b467d1ddfceafb
+github.com/containerd/console cb7008ab3d8359b78c5f464cb7cf160107ad5925
 github.com/containerd/containerd 08f7ee9828af1783dc98cc5cc1739e915697c667
 github.com/containerd/continuity d8fb8589b0e8e85b8c8bbaa8840226d0dfeb7371
-github.com/coreos/etcd v3.2.1
+github.com/coreos/etcd v3.3.9
 github.com/cpuguy83/go-md2man v1.0.8
-github.com/davecgh/go-spew 346938d642f2ec3594ed81d874461961cd0faa76
+github.com/davecgh/go-spew 346938d642f2ec3594ed81d874461961cd0faa76 # v1.1.0
```

Comment:
Why not using the tag directly?

Reply:
Ah! Yes, there's been many discussions about those. Ideally, there'd be a way to specify _both_ tag _and_ commit/sha;

reason to use a tag:

- it's human-readable
- it's a release done by the upstream dependency

reason to _not_ use a tag:

- we've had upstreams re-tagging (discovered when `vendor check` failed and no changes were made in the vendor.conf)
- we've had cases where it turned out not to be a _tag_ but a _branch_ named `v1.0` (but we _thought_ it was a tag), then found out about that (again) when `vendor check` failed.
- a commit-sha is unique, and can never change


So... how to get the best of both worlds?

- use sha
- leave comment what version it should match with?

It's tricky; because we also have to verify that the commit matches the tag 😞 possibly the `git describe` notation would be an option, e.g.;

`v0.2.0-7-g1817cd4`

- 7 commits since `v0.2.0`
- sha is `1817cd4`

not sure if that actually _verifies_ if they match though? 🤔

Reply:
Oh right, lot's of issues I wasn't aware of 😓 Then that's ok for me 😸